pkg/volume/git_repo/git_repo_test.go

```diff
@@ -95,13 +102,24 @@ func testSetUp(plug volume.VolumePlugin, builder volume.Builder, t *testing.T) {
 	if !reflect.DeepEqual(expectedCmds, fcmd.CombinedOutputLog) {
 		t.Errorf("unexpected commands: %v, expected: %v", fcmd.CombinedOutputLog, expectedCmds)
 	}
-	expectedDirs := []string{g.GetPath(), g.GetPath() + "/kubernetes", g.GetPath() + "/kubernetes"}
+
+	repoDir := ""
+	if g.target != "." {
+		repoDir = "/kubernetes"
+	}
+	expectedDirs := []string{g.GetPath(), g.GetPath() + repoDir, g.GetPath() + repoDir}
 	if len(fcmd.Dirs) != 3 || !reflect.DeepEqual(expectedDirs, fcmd.Dirs) {
 		t.Errorf("unexpected directories: %v, expected: %v", fcmd.Dirs, expectedDirs)
 	}
 }
 
 func TestPlugin(t *testing.T) {
+	testPluginWithTarget("target_dir", t)
```

Comment:
You have 3 tests within your single test func.

If you are testing many scenarios, a common practice is to make a map of them, like so:  https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation_test.go#L252